encoding/value/decimal_value.rs

```diff
@@ -214,6 +226,53 @@ macro_rules! impl_ref_ops {
 
 impl_ref_ops! { Add::add, Sub::sub, Mul::mul }
 
+impl Div<u64> for Decimal {
+    type Output = Self;
+
+    fn div(self, rhs: u64) -> Self::Output {
+        let i128 = (self.integer as i128 * FRACTIONAL_PART_DENOMINATOR as i128 + self.fractional as i128) / rhs as i128;
+        let int = i128.div_euclid(FRACTIONAL_PART_DENOMINATOR as i128) as i64;
+        let frac = i128.rem_euclid(FRACTIONAL_PART_DENOMINATOR as i128) as u64;
+        Decimal::new(int, frac)
+    }
+}
```

Comment:
Just checking my understanding here in contrast to the Std operator:

i128 `/` does integer division and throws off the decimal. So we'd have 
` (int * DENOM + frac )/DENOM + (truncated_decimal /  DENOM)`
Since there's no loss of precision beyond the digits we wouldn't be able to represent anyway, we can keep it in a `Decimal`

But with sqrt, we pass through f64 so we have a loss of precision that's much worse than what `Decimal` otherwise guarantees?